x-pack/plugins/security_solution/public/overview/components/detection_response/hooks/use_navigate_to_timeline.tsx

```diff
@@ -63,10 +65,17 @@ export const useNavigateToTimeline = () => {
     const mainFilter = entityFilters.shift();
 
     if (mainFilter) {
-      const dataProvider = getDataProvider(mainFilter.field, '', mainFilter.value);
+      const dataProvider = getDataProvider(
+        mainFilter.field,
+        uuid.v4(),
```

Comment:
why the uuid?

Reply:
needed something unique for each data provider. Without unique IDs the UI acts in a weird behaviour. I considered using the variables I already had access to to make something unique, but was worried that a user could technically make the same query or have the same filters multiple times.- A technicality to said technicality :P is that this is all in an internal hook and with correct implementation should not be a problem. 🤔 Should I change it?

Reply:
That's fair, we have used a combination of field values and context information for the unique id. The only way to generate something unique here is if you included the type operator, but even then a user could accidentally duplicate a field with the same operator and run into that issue. I wonder if we could use the hooks `useId` [ref](https://reactjs.org/docs/hooks-reference.html#useid) here, and also potentially use that in place of all the places were we make really contrived dataProvider id's?

Reply:
what the heck! I didn't know React gave us that outta the box. Thanks, changed!

Reply:
No wait...thats as of version 18 :P We're still using 17 :(